preliz/internal/optimization.py

```diff
@@ -452,3 +455,9 @@ def func(x, dist, q):
         left, right = right, right * factor
 
     return brentq(func, left, right, args=(dist, q))
+
+
+def get_weighted_rvs(targets, weights, size, seed):
+    target_rnd_choices = np.random.choice(len(targets), size=size, p=weights)
+    samples = [target.rvs(size, np.random.default_rng(seed)) for target in targets]
+    return np.choose(target_rnd_choices, samples)
```

Comment:
create a random generator once and then use it everywhere. I think it should be generated inside ppe.py and then propagated as needed.

Reply:
For this, I think rng needs to be defined in `optimize_pymc_model`...

Reply:
Ok but the user should be able to pass a seed to ppe

Reply:
Hmm...
This sounds good, we can allow the user to input a seed keeping default to be 0.